Improvement/api variables (#762)
codemagic-ci-cd/codemagic-docs: content/rest-api/applications.md

  }' 'https://api.codemagic.io/apps/<app_id>/variables'
```

> #### Example adding file variables

Reviewer: ```suggestion
#### Example of adding file variables

```

> Things are a bit more complicated when adding files to variables due to how `JSON` handles newlines and environment variables.

Reviewer: IMO this paragraph is not necessary here

> It is possible to pass text-based files to the `cURL` command with the help of CLI tools such as `sed` or `awk` (which is

Reviewer: ```suggestion
It is possible to pass text-based files to the `cURL` command with the help of CLI tools, such as `sed` or `awk` (which is

```

used in the example below). These tools provide options to properly retain newlines, which are essential for some files

> (e.g., private keys) to function correctly.

Reviewer: ```suggestion
(e.g. private keys) to function correctly.

```

```bash
FILE=$(awk 1 ORS='\\n' file_path)
curl -XPOST -H 'x-auth-token: <API TOKEN>' -H 'Content-Type: application/json;charset=utf-8' -d "{
    \"key\":\"FOO\",
    \"value\":\"$FILE\", 
    \"group\":\"production\", 
    \"secure\": true
    }" 'https://api.codemagic.io/apps/<app_id>/variables'
```

> To add binary-based files (e.g., images), they need to be 

Reviewer: ```suggestion
To add binary-based files (e.g. images), they need to be 

```

[`base64 encoded`](../variables/environment-variable-groups/#storing-sensitive-valuesfiles) first before passing them

> to the value parameter as strings. Note that they will also have to be `base64 decoded` during the build to use.

Reviewer: ```suggestion
to the value parameter as strings. Note that the values will have to be `base64 decoded` during the build so as to use them.

```
